daft/expressions/expressions.py

```diff
@@ -2222,6 +2222,19 @@ def count_matches(
 
         return count_matches(self, patterns, whole_words=whole_words, case_sensitive=case_sensitive)
 
+    def regexp_count_matches(
```

Comment:
Thoughts on naming it `regexp_count`? That might be more consistent with [PySpark](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.regexp_count.html#pyspark.sql.functions.regexp_count) and some SQL dialects (e.g. [snowflake](https://docs.snowflake.com/en/sql-reference/functions/regexp_count), [oracle](https://docs.oracle.com/en/database/oracle/oracle-database/23/sqlrf/REGEXP_COUNT.html))